apps/allocations/app/components/App/Payouts.js

```diff
@@ -0,0 +1,150 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import styled from 'styled-components'
+import {
+  Table,
+  TableHeader,
+  TableRow,
+  TableCell,
+  Text,
+  Button,
+  IdentityBadge,
+  Viewport,
+  theme,
+  IconCheck,
+  IconCross,
+  IconFundraising,
+  IconTime,
+  ContextMenu,
+  ContextMenuItem,
+} from '@aragon/ui'
+import { displayCurrency } from '../../utils/helpers'
+import {
+  PayoutDescription,
+  PayoutsTable,
+  NarrowList,
+  NarrowListPayout,
+  AmountBadge,
+} from './PayoutsTables'
+import { Empty } from '../Card'
+import { provideNetwork } from '../../../../../shared/ui'
+
+const translateToken = (token) => {
+  if (token == 0x0) {
+    return 'ETH'
+  }
+}
+
+const PayoutStatusWrapper = ({ color, icon, title, posTop = 0 }) => {
+  const Icon = icon
+  return (
+    <PayoutStatus color={color}>
+      <Icon style={{ position: 'relative', top: posTop, marginRight: '10px' }} />
+      {title}
+    </PayoutStatus>
+  )
+}
+
+const showStatus = ({ distSet, recurring, startTime }) => {
+  let status
+  if (!recurring) {
+    if (distSet) status = 1
+    else if (!distSet) status = 2
+    else status = 3
+  }
+  else {
+    if(Date.now() < startTime) status = 0
+    else status = 1
+  }
+  switch(status) {
+  case 0: return <PayoutStatusWrapper title="Distribution in Progress..." icon={IconTime} color={theme.textSecondary} posTop={1} />
```

Comment:
Love this code pattern, can we set this returns as constants or something so this function is cleaner?